Initial pull request

Artem-Sotnikov opened the pull request:
This analyzer takes csv data (and optionally, short descriptors of that data), and outputs one (or several) excel spreadsheets with the data analyzed. The code has been tested on the existing recovery dataset and works correctly. It's being uploaded to github for future reference purposes. 

Any code efficiency or style suggestions are welcome!

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1)
<!-- Reviewable:end -->

Artem-Sotnikov committed:
Add all files, including those used in testing.

Artem-Sotnikov committed:
Fix gitignore, remove excel and csv files

jlsajfj commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+
+        names.append(DName('TARGET_COEFF_DRAG', attr_text=self.TARGET_COEFF_DRAG))
+
+def f_create_constants_table(target_sheet, loc=None):
```
should all of these constants_table values (ie "=B1/0.3048") be based on the constants defined earlier?

jlsajfj commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+
+    target_sheet[Consts.L_NOM_DIAM_TITLE] = "Nominal diamater of parachute (m | ft)"
+    target_sheet[Consts.L_NOM_DIAM_M] = 4.965
+    target_sheet[Consts.L_NOM_DIAM_FT] = "=B1/0.3048"
```
For example "=" + L_NOM_DIAM_M + "/0.3048"

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+        columns.
+    '''
+    
+    if cols:
```
So, what you are doing here is checking if there is some number of columns (other than 0) and if so doing a for loop, other wise doing a very similar while loop.

The code in these two loops is very similar, and it would be nice if it was just one loop, so we can avoid re-writing code.

what if we cols default to `cols = float("inf")` and then we could just make the condition in the while loop be
`while target_sheet.cell(1, col_count).value and col_count <= cols:`

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+                    if curr_width > max_width:
+                        max_width = curr_width
+
+            target_sheet.column_dimensions[get_column_letter(
```
Spacing looks odd, there is this tool the software team uses for formatting code called `format.sh`, the executable is in the topside repo, but I can always just send it to you and you can add it to this.

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+import csv
+from openpyxl.styles.alignment import Alignment
+
+# Configuration Variables:
```
We could put these in their own file called `ConfigVariables.py` and then import that at the top of the program.

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+            col_count += 1
+
+
+class Consts():
```
I find it odd to create a class that never has any instants created of it. Here is a better way to handle this.

So you have a bunch of constants like L_NOM_DIAM_TITLE, which each have a spreadsheet location, an actual name and a value that is going to end up in the table (I believe that is it). Why not create a class called `Constant` which has a name field, a location field and a table value field. We can create one of these for each of the constants and store them in a dict, where the keys are the names.

The advantage of this is that we will be able to loop through the constants, but also we can access each on individually. Below there will be more discussion about why this is advantageous

KavinSatheeskumar replied:
Also, put this dict in another file and import it

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+        """
+        names = target_workbook.defined_names # Used to shorten lines 
+        
+        names.append(DName('NOM_DIAM_M', attr_text=self.NOM_DIAM_M))
```
Here is the first example of why it would be advantageous of putting all those constants into a map would be useful.
Let our dict be called `const_dict`, that let's us do this

`
for const_name, const_obj in const_dict.items:
    names.append(Dname(const_name, attr_text=const_obj.spreadsheet_location`

^^there should be some newlines in there, but the code thing here is removing them all

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+        argument.
+    """
+
+    target_sheet[Consts.L_NOM_DIAM_TITLE] = "Nominal diamater of parachute (m | ft)"
```
This is the second example of why that dictionary would be useful, all this code could be done as
`
for const_obj in const_dict:
    target_sheet[const_obj.spreadsheet_location] = const_obj.table_value
`

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+                (openpyxl starts counting at 1 so this is the true first row in the spreadsheet)
+    
+    """
+    sheet.cell(target_row, 1, "Time (ms)")
```
Might be worth storing these in a list and looping through, its your call, not very important

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+
+    ws_average_window = 0
+    if row_idx - ws_average_window >= 1:
+        averaged_ws_lower_bound = row_idx - ws_average_window
```
again, not a big issue, but
`averaged_ws_lower_bound = max(row_ids - ws_average_window, 1)`

In general using max to do stuff like this can hinder readability, but I feel like this is a case where it is small enough that its fine

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+
+    drag_area_chart.append(drag_area_series)
+
+    if DEBUG_MODE_VERBOSE:
```
This is the only use of DEBUG_MODE_VERBOSE, is it really necessary here?

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+    if DEBUG_MODE_VERBOSE:
+        print(data_sheet.cell(max_idx - 1, 1).value)
+
+    chart.x_axis.scaling.max = data_sheet.cell(max_idx - 1, 1).value
```
Might wanna make some of this stuff optional parameters

KavinSatheeskumar replied:
or config constants

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+    drag_area_chart += chart
+    if output_sheet == None:
+        output_sheet = data_sheet
+        output_sheet.add_chart(drag_area_chart, 'M2')
```
Since this is the last time the output_sheet variable is used, you could go `data_sheet.add_chart(...)`

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+
+    row_idx = 1
+    for row in list(csvreader):
+        if row_idx and row_idx != 1:
```
`If row_idx > 1`

KavinSatheeskumar commented on truck_test_rapid_analyzer.py:
```diff
@@ -0,0 +1,614 @@
+        space_columns(ws_data, 1)
+        space_columns(ws_consts, 10, 1)
+
+        active_wb.save(otarget)
```
Might be nice to try and break the code up into smaller individual files and have them import each other. Jacob told me that you want the code in main to be as small as possible and everything else should be in its own little piece.

Over all though, your code was nice and very easy to review! Thanks for making it well written and easy to understand

Artem-Sotnikov committed:
Create improved version

Add several features, including better constants handling, examples, better documentation, command-line variable passing, more robust main method.

Artem-Sotnikov committed:
Begin cleanup & add unit test

Artem-Sotnikov committed:
Complete cleanup

KavinSatheeskumar reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1)* status: 0 of 25 files reviewed, 18 unresolved discussions (waiting on @Artem-Sotnikov and @wendi-yu)

___
*[`constants.py` line 121 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jvH3E664zzkPTMatT:-N6jvH3E664zzkPTMatU:b-nqblm6) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/constants.py#L121)):*
> ```Python
>              Defaul value is None for same reason as varname_conv. 
>         """
>         self.variables_raw.append(locals())  # Add raw variable input to class variables list
> ```

Not entirely sure what this is, I highly doubt you should be using `locals()` for whatever this is and this variable `self.variables_raw` is never used again
___
*[`tests/rapid_analyzer_example_direct.py` line 8 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jwnn67eOWjp5QPh5W:-N6jwnn67eOWjp5QPh5X:b3gwyre) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/tests/rapid_analyzer_example_direct.py#L8)):*
> ```Python
>     from ..truck_test_rapid_analyzer import execute_complete_analysis
> except:
>     # Hack-ey workaround that allows this to run using eclipse IDE
> ```

Hottest of hot takes, why are you running using eclipse IDE?
___
*[`tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv` line 1 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jwPkW5g1nGoRVBCtN:-N6jwPkW5g1nGoRVBCtO:bj1elr8) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv#L1)):*
> ```CSV
> Timestamp,ANEMOMETER,LOAD CELL
> ```

Can we git ignore this file?


<!-- Sent from Reviewable.io -->

Artem-Sotnikov reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1)* status: 0 of 25 files reviewed, 18 unresolved discussions (waiting on @jlsajfj, @KavinSatheeskumar, and @wendi-yu)

___
*[`constants.py` line 121 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jvH3E664zzkPTMatT:-N6k2j1m2N-J0_bDDfrr:b-20w5d0) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/constants.py#L121)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Not entirely sure what this is, I highly doubt you should be using `locals()` for whatever this is and this variable `self.variables_raw` is never used again
</blockquote></details>

This enables the class to maintain a log of every variable that is added. While I didn't end up using this anywhere in the code I think this isn't a bad idea for extending outher functionality. This would allow a later class to iterate over the constants that were created.
___
*[`truck_test_rapid_analyzer.py` line 14 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-14:-N6k36wY46F3itboGI-Y:b6m399s) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L14)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

We could put these in their own file called `ConfigVariables.py` and then import that at the top of the program.
</blockquote></details>

Yeah that's probably a cleaner way of doing this. 
___
*[`truck_test_rapid_analyzer.py` line 48 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-48:-N6k3hwvB7g2uk1VL8D8:ba7l6ic) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L48)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

So, what you are doing here is checking if there is some number of columns (other than 0) and if so doing a for loop, other wise doing a very similar while loop.

The code in these two loops is very similar, and it would be nice if it was just one loop, so we can avoid re-writing code.

what if we cols default to `cols = float("inf")` and then we could just make the condition in the while loop be
`while target_sheet.cell(1, col_count).value and col_count <= cols:`
</blockquote></details>

Sorry, I didn't clarify this back then but I don't actually see the redundancy. The while loop is for columns, while the for loop is for rows (depth). The reason why I made the outer loop a while is because it is allowed two exit conditions (limit reached or no more filled cells) while the for loop must scan to the specified depth. It may be slightly better to make both of the whiles but I really doubt that the inner loop will need to make use of this functionality. 
___
*[`truck_test_rapid_analyzer.py` line 78 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-78:-N6kLCAs3eMPXDWb7DfQ:bvm6smh) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L78)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Spacing looks odd, there is this tool the software team uses for formatting code called `format.sh`, the executable is in the topside repo, but I can always just send it to you and you can add it to this.
</blockquote></details>

Yah, I'll run format.sh on this thing in the nextcommit.
___
*[`truck_test_rapid_analyzer.py` line 85 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-85:-N6k4iGLCXUao5CQRVfd:bxc869j) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L85)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Also, put this dict in another file and import it
</blockquote></details>

Again, my apologies for not clarifying this earlier. This is not related to constants in any way, but rather this formats the headers in the various calculation scripts. This could in all honesty be a function by itself with a bunch of constants defined inside it or imported from elsewhere as you suggested. However, I feel that having it be in a class is more extensible for future functionality, and allows the constants defined inside to be used elsewhere. I'm not sure if moving it to a separate file will improve the readability, but it can be done.  The actual constants are handled in a similar way to what you describe. 
___
*[`truck_test_rapid_analyzer.py` line 172 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-172:-N6k69PW-ojh40eXCQIG:b81nrd8) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L172)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Here is the first example of why it would be advantageous of putting all those constants into a map would be useful.
Let our dict be called `const_dict`, that let's us do this

`
for const_name, const_obj in const_dict.items:
    names.append(Dname(const_name, attr_text=const_obj.spreadsheet_location`

^^there should be some newlines in there, but the code thing here is removing them all
    
</blockquote></details>

I am hesitant to do this because if constants change it would require changes elsewhere in the code. The scenario you describe has already been abstracted away in the rewritten constants class.
___
*[`truck_test_rapid_analyzer.py` line 196 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-196:-N6kLgQ11jMKv2uYOY-6:b-w7tnx2) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L196)):*
<details><summary><i>Previously, jlsajfj wrote…</i></summary><blockquote>

should all of these constants_table values (ie "=B1/0.3048") be based on the constants defined earlier?
</blockquote></details>

This has now been corrected. Now all mathematics are performed correctly.
___
*[`truck_test_rapid_analyzer.py` line 210 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-210:-N6k6n_O2UA_DvN6paNr:b5manlp) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L210)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

This is the second example of why that dictionary would be useful, all this code could be done as
`
for const_obj in const_dict:
    target_sheet[const_obj.spreadsheet_location] = const_obj.table_value
`
</blockquote></details>

These are not constants but headers in the calculation table, which are distinct.
___
*[`truck_test_rapid_analyzer.py` line 212 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-212:-N6kLpGe9umpmpyRm_RE:b-q64pkt) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L212)):*
<details><summary><i>Previously, jlsajfj wrote…</i></summary><blockquote>

For example "=" + L_NOM_DIAM_M + "/0.3048"
</blockquote></details>

See above.
___
*[`truck_test_rapid_analyzer.py` line 262 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-262:-N6k6yFSBpeqQ60BB3C-:b-6irc9t) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L262)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Might be worth storing these in a list and looping through, its your call, not very important
</blockquote></details>

Given the function of the class as a way to keep coherence between headers and other functions using the data, I don't think that adding a list on top will take this significantly cleaner. 
___
*[`truck_test_rapid_analyzer.py` line 333 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-333:-N6k7NRG-Iw81dA9YYZW:b-wa3zav) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L333)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

again, not a big issue, but
`averaged_ws_lower_bound = max(row_ids - ws_average_window, 1)`

In general using max to do stuff like this can hinder readability, but I feel like this is a case where it is small enough that its fine
</blockquote></details>

Yeah this is a little opaque, I think breaking it up into distinct branches is good enough.
___
*[`truck_test_rapid_analyzer.py` line 426 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-426:-N6kHjS-2KyHzeM0MeJU:b694uer) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L426)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

This is the only use of DEBUG_MODE_VERBOSE, is it really necessary here?
</blockquote></details>

I think so, this adds a lot to the printout. Also I think that in general, two levels of debug messages is a good way of structuring the debug of the code.
___
*[`truck_test_rapid_analyzer.py` line 429 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-429:-N6k7UZ_CxvTzQ2OSEnA:b-3n8xts) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L429)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

or config constants
</blockquote></details>

I think this would bloat the config way too much.
___
*[`truck_test_rapid_analyzer.py` line 449 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-449:-N6kIOAi6wQ5ZyzSfs7U:b-fd83m2) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L449)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Since this is the last time the output_sheet variable is used, you could go `data_sheet.add_chart(...)`
</blockquote></details>

That is true, but I think that readability is improved by explicitly setting the original data sheet as the output destination for the graph. I'll add a comment to make this clearer.
___
*[`truck_test_rapid_analyzer.py` line 512 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-512:-N6kIZfK6RkV_JMKXyoS:b5jdlqb) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L512)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

`If row_idx > 1`
</blockquote></details>

Whoops yep that's an L
___
*[`truck_test_rapid_analyzer.py` line 614 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-614:-N6kJX2DE6NpyPNcppz7:b4u682n) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L614)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Might be nice to try and break the code up into smaller individual files and have them import each other. Jacob told me that you want the code in main to be as small as possible and everything else should be in its own little piece.

Over all though, your code was nice and very easy to review! Thanks for making it well written and easy to understand
</blockquote></details>

Broadly speaking, I think that may be right. As part of a future rewrite, I think this is worth including, but I think this is out of the scope of the current PR.
___
*[`tests/rapid_analyzer_example_direct.py` line 8 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jwnn67eOWjp5QPh5W:-N6kKtl6CbEkPvSl5xJ0:blsb2xw) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/tests/rapid_analyzer_example_direct.py#L8)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Hottest of hot takes, why are you running using eclipse IDE?
</blockquote></details>

Just humor me
___
*[`tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv` line 1 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jwPkW5g1nGoRVBCtN:-N6kKwKQ0xT0xs03cnSd:bq7ft1r) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv#L1)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Can we git ignore this file?
</blockquote></details>

No, this is a sample file for running the example bat scripts and for providing an example of what kind of files the calculator is meant to analyze. This is a feature of this code package.


<!-- Sent from Reviewable.io -->

Artem-Sotnikov committed:
Address PR comments, update Readme

Specifically, the hard-coded configs were pulled into a separate file for cleanliness. A few clarifying comments were added and a few minor inefficiencies corrected.

KavinSatheeskumar reviewed (changes requested):
> Reviewed 5 of 25 files at r2, 3 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1)* status: 8 of 26 files reviewed, 11 unresolved discussions (waiting on @Artem-Sotnikov, @jlsajfj, @KavinSatheeskumar, and @wendi-yu)

___
*[`config_variables.py` line 1 at r3](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6xIK5k7Pj2J3zqQXS5:-N6xIK5l3Xawuu2VV3Jz:b-rulnu2) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/820027c4cc10701581671982087cae28dd74eef6/config_variables.py#L1)):*
> ```Python
> # Configuration Variables:
> ```

Maybe a future note, but might be worth thinking about what the best way to config stuff is. However, I think this is probably more than fine
___
*[`constants.py` line 121 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jvH3E664zzkPTMatT:-N6xIVjJAbwx04UPqU2C:bixo9eh) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/constants.py#L121)):*
<details><summary><i>Previously, Artem-Sotnikov wrote…</i></summary><blockquote>

This enables the class to maintain a log of every variable that is added. While I didn't end up using this anywhere in the code I think this isn't a bad idea for extending outher functionality. This would allow a later class to iterate over the constants that were created.
</blockquote></details>

Yeah, that is now what this does, `locals()` produces a list of every local variable at the time. If you are calling this function in a for loop, with counter `i`, you'll add `i` to your list. Also, it returns a dictionary of names and values. so what you will have is an ever growing list of dictionaries, mainly consisting of things that are not relevant.
___
*[`truck_test_rapid_analyzer.py` line 85 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-85:-N6xLOfG9GUDMUYnvTFa:b-nyoew7) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L85)):*
<details><summary><i>Previously, Artem-Sotnikov wrote…</i></summary><blockquote>

Again, my apologies for not clarifying this earlier. This is not related to constants in any way, but rather this formats the headers in the various calculation scripts. This could in all honesty be a function by itself with a bunch of constants defined inside it or imported from elsewhere as you suggested. However, I feel that having it be in a class is more extensible for future functionality, and allows the constants defined inside to be used elsewhere. I'm not sure if moving it to a separate file will improve the readability, but it can be done.  The actual constants are handled in a similar way to what you describe. 
</blockquote></details>

See, its just that there is something you do pretty frequently in the code that I might wanna try and simplify. 

Often times, you have like a bunch of constants (here the columns of the table) and you do something relatively similar for each one. I was trying to find a way to strip this down to make it a bit less space consuming. This way, if we ever add a new header or something, we don't have to change as much code. That being said, I think you'd rather keep it like this for readability reasons based on your other comments, which is fair enough. However, at some point, we should find a way to reduce doing stuff like that, solely just to make it easier to modify this code. 

I think this kinda harkens back to the general config problem, as a lot of this is very dependent on the specific thing you are doing, making it hard to generalize.
___
*[`truck_test_rapid_analyzer.py` line 429 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-429:-N6xKbXV9nhHuh3GQWoo:b-a4o1dl) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L429)):*
<details><summary><i>Previously, Artem-Sotnikov wrote…</i></summary><blockquote>

I think this would bloat the config way too much.
</blockquote></details>

Fair, we can think about how we wanna handle config and such in later PRs


<!-- Sent from Reviewable.io -->

KavinSatheeskumar reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1)* status: 8 of 26 files reviewed, 9 unresolved discussions (waiting on @Artem-Sotnikov, @jlsajfj, @KavinSatheeskumar, and @wendi-yu)

___
*[`tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv` line 1 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jwPkW5g1nGoRVBCtN:-N6xMalUAuaayHH1-NU_:b-z7xqvf) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv#L1)):*
<details><summary><i>Previously, Artem-Sotnikov wrote…</i></summary><blockquote>

No, this is a sample file for running the example bat scripts and for providing an example of what kind of files the calculator is meant to analyze. This is a feature of this code package.
</blockquote></details>

Fair, still makes it a bit of a pain to read the code. Should probably find some way of dealing with the test, but for now its not an issue.


<!-- Sent from Reviewable.io -->

Artem-Sotnikov reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1)* status: 8 of 26 files reviewed, 9 unresolved discussions (waiting on @jlsajfj, @KavinSatheeskumar, and @wendi-yu)

___
*[`config_variables.py` line 1 at r3](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6xIK5k7Pj2J3zqQXS5:-N6xRQV76b5XIPPle_1h:bu7le0i) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/820027c4cc10701581671982087cae28dd74eef6/config_variables.py#L1)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Maybe a future note, but might be worth thinking about what the best way to config stuff is. However, I think this is probably more than fine
</blockquote></details>

There is already a way to pass configuration via YAML by command line, which is I think a pretty good way of handling this. These variables are an alternative backup way of passing config.
___
*[`constants.py` line 121 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jvH3E664zzkPTMatT:-N6xRm2N71kFxPF2yK_E:baxjmpf) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/constants.py#L121)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Yeah, that is now what this does, `locals()` produces a list of every local variable at the time. If you are calling this function in a for loop, with counter `i`, you'll add `i` to your list. Also, it returns a dictionary of names and values. so what you will have is an ever growing list of dictionaries, mainly consisting of things that are not relevant.
</blockquote></details>

Unless I am very mistaken about the function of locals, by making this call right at the beginning of the function, it will merely save the arguments passed into it, which is the desired functionality I was going for. Perhaps I should add a comment to ensure that absolutely nothing else gets put before it? 
___
*[`truck_test_rapid_analyzer.py` line 85 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-85:-N6xV6Aa-i_TnzGFieIc:bhivng3) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L85)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

See, its just that there is something you do pretty frequently in the code that I might wanna try and simplify. 

Often times, you have like a bunch of constants (here the columns of the table) and you do something relatively similar for each one. I was trying to find a way to strip this down to make it a bit less space consuming. This way, if we ever add a new header or something, we don't have to change as much code. That being said, I think you'd rather keep it like this for readability reasons based on your other comments, which is fair enough. However, at some point, we should find a way to reduce doing stuff like that, solely just to make it easier to modify this code. 

I think this kinda harkens back to the general config problem, as a lot of this is very dependent on the specific thing you are doing, making it hard to generalize.
</blockquote></details>

This ultimately boils down to style. By having stuff living in some lists or dicts somewhere else instead of directly in the function calls, the number of lines is not reduced. The code becomes marginally more elegant at the exchange of some readability. Having everything in a single function call, in my opinion, is easier to modify than fiddling with a bunch of different lists, but this is just my opinion. 
___
*[`truck_test_rapid_analyzer.py` line 429 at r1](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N0mz4L4tb-gffaZTfWP-r1-429:-N6xY0QhBuBuOI6PL28o:b-wsj4cf) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/d2183798e822a762ae385a92d1b57e2fc99554d8/truck_test_rapid_analyzer.py#L429)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Fair, we can think about how we wanna handle config and such in later PRs
</blockquote></details>

Yeah, I think having a separate config for the graphs may be potentially worthwhile. 
___
*[`tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv` line 1 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jwPkW5g1nGoRVBCtN:-N6xYc1lD7cSJFb07OP2:b7f36un) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/tests/example_data_directory/sample_csv_of_log_2021-10-10_0.15tau.csv#L1)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Fair, still makes it a bit of a pain to read the code. Should probably find some way of dealing with the test, but for now its not an issue.
</blockquote></details>

I'm still a bit confused here. What code are you referring to?


<!-- Sent from Reviewable.io -->

KavinSatheeskumar reviewed (changes requested):
> Reviewed 14 of 25 files at r2, 4 of 7 files at r3.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1)* status: all files reviewed, 3 unresolved discussions (waiting on @Artem-Sotnikov, @jlsajfj, and @wendi-yu)

___
*[`constants.py` line 121 at r2](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-N6jvH3E664zzkPTMatT:-N800Ra6CklWxMIEV339:bteg9w0) ([raw file](https://github.com/waterloo-rocketry/truck-test-rapid-analyzer/blob/006ced726155651f7b234655300707d2348df90c/constants.py#L121)):*
<details><summary><i>Previously, Artem-Sotnikov wrote…</i></summary><blockquote>

Unless I am very mistaken about the function of locals, by making this call right at the beginning of the function, it will merely save the arguments passed into it, which is the desired functionality I was going for. Perhaps I should add a comment to ensure that absolutely nothing else gets put before it? 
</blockquote></details>

You are right, I misunderstood some semantics.

Still, don't do it this way, it raises questions about the semantics of the `locals()` function in python and greatly hinders readability. In my opinion, you should just manually create the dictionary object and add it


<!-- Sent from Reviewable.io -->

Artem-Sotnikov committed:
Resolve the locals call in consants with more readable code

The previosu solution was kind of leaning into code golf, it was hard to read what was going on and the comments weren't too helpful either. Especially since this isn't even part of the core functionality, it shouldn't have been written like this. A much more simple solution that's much clearer and only like one line longer has been implemented.

Artem-Sotnikov committed:
Formatting pass

KavinSatheeskumar reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/truck-test-rapid-analyzer/1#-:-NBdnbUq5pnc7WU78oav:bnfp4nl)* status: 25 of 26 files reviewed, 2 unresolved discussions (waiting on @jlsajfj, @KavinSatheeskumar, and @wendi-yu)



<!-- Sent from Reviewable.io -->
